numpyro/distributions/discrete.py

```diff
@@ -82,7 +82,8 @@ def __init__(self, probs, validate_args=None):
         super(BernoulliProbs, self).__init__(batch_shape=jnp.shape(self.probs), validate_args=validate_args)
 
     def sample(self, key, sample_shape=()):
-        return random.bernoulli(key, self.probs, shape=sample_shape + self.batch_shape)
+        samples = random.bernoulli(key, self.probs, shape=sample_shape + self.batch_shape)
+        return samples.astype(canonicalize_dtype(jnp.int64))
```

Comment:
For clarity, this will only coerce to int64 if the corresponding flag is set but by default this should use int32, right? If that's the case, I wonder if there is a more idiomatic way of doing this in JAX. This can be a source of confusion for anyone new to the codebase.

Reply:
Looking at [jax.numpy](https://github.com/google/jax/blob/master/jax/_src/numpy/lax_numpy.py) source code, it seems that we can use
```
samples.astype(jnp.result_type(x, int))
```
Does this look better to you?

Reply:
That seems fine - basically a `jnp.array(True) * 0` will give result in a `int32` type. Can we use this pattern?

Reply:
Sure! :) I'll make that change.